integration-tests/byoid/integration_test.go

```diff
@@ -0,0 +1,206 @@
+// Copyright 2021 Google LLC.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// +build integration
+
+// To run this test locally, you will need to do the following:
+// • Navigate to your Google Cloud Project
+// • Get a copy of the Service Account Key File from somebody
+// • If you are unable to obtain an existing key file, create one:
+//    • > IAM and Admin > Service Accounts
+//    • Under the needed Service Account > Actions > Manage Keys
+//    • Add Key > Create New Key
+//    • Select JSON, and the click Create
+// • > Compute > Compute Engine > VM Instances
+// • Look for an available VM Instance, or create one
+// • On the VM Instance, click the SSH Button
+// • Upload your Service Account Key File
+// • Upload this script, along with setup.sh
+// • Get a copy of the needed environment variables from somebody, and upload those too
+// • Set your environment variables (Usually this will be `source env.conf`)
+// • If the setup script has not yet been run, then run it
+// • `go test -tags integration`
+
+package byoid
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"testing"
+
+	"google.golang.org/api/dns/v1"
+	"google.golang.org/api/idtoken"
+	"google.golang.org/api/option"
+)
+
+const (
+	envCredentials  = "GCLOUD_TESTS_GOLANG_KEY"
+	envAudienceOIDC = "GCLOUD_TESTS_GOLANG_AUDIENCE_OIDC"
+	envProject      = "GCLOUD_TESTS_GOLANG_PROJECT_ID"
+)
+
+var (
+	oidcAudience string
+	oidcToken    string
+	awsToken     string
+	clientID     string
+	projectID    string
+)
+
+// TestMain contains all of the setup code that needs to be run once before any of the tests are run
+func TestMain(m *testing.M) {
+	keyFileName := os.Getenv(envCredentials)
+	if keyFileName == "" {
+		log.Fatalf("Please set %s to your keyfile", envCredentials)
+	}
+
+	projectID = os.Getenv(envProject)
+	if projectID == "" {
+		log.Fatalf("Please set %s to the ID of the project", envProject)
+	}
+
+	oidcAudience = os.Getenv(envAudienceOIDC)
+	if oidcAudience == "" {
+		log.Fatalf("Please set %s to the OIDC Audience", envAudienceOIDC)
+	}
+
+	clientID = getClientID(keyFileName)
+	oidcToken = generateGoogleToken(keyFileName)
+
+	// This line runs all of our individual tests
+	os.Exit(m.Run())
+}
+
+// keyFile is a struct to extract the relevant json fields for our ServiceAccount KeyFile
+type keyFile struct {
+	ClientEmail string `json:"client_email"`
+	ClientID    string `json:"client_id"`
+}
+
+func getClientID(keyFileName string) string {
+	kf, err := os.Open(keyFileName)
+	if err != nil {
+		log.Fatalf("Failed to open file '%s': %v", keyFileName, err)
```

Comment:
nit: for both of these helper functions for main return the erorrs and log fatal from there.

Reply:
done